### PR TITLE
Added minhash encoder and the Gamma-Poisson factorization. Also added…

### PR DESCRIPTION
**Pull request by alexis-cvetkov**
… a reformat_input method for similarity, minhash and gamma-poisson encoders so that input lists, arrays and pandas objects can be used in fit_transform.

### CODE REVIEW

File: dirty_cat/minhash_encoder.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                   "minmax_hash not implemented with murmur"
+            return self.minhash(
+                    string, n_components=self.n_components,
+                    ngram_range=self.ngram_range)
+        else:
+            raise ValueError("hashing function must be 'fast' or"
+                             "'murmur', got '{}'"
+                             "".format(self.hashing))
+
+    def reformat_input(self, X):
```
Comment (GaelVaroquaux) [inline dirty_cat/minhash_encoder.py]: This blank line shouldn't be there.

File: dirty_cat/gamma_poisson_factorization.py [hunk 2]
```diff
@@ -0,0 +1,478 @@
+import numpy as np
+
+import pandas as pd
```
Comment (jeromedockes) [inline dirty_cat/gamma_poisson_factorization.py]: if you need pandas add it to the `install_requires` list in setup.py to fix the CI error

**Comment by GaelVaroquaux** [comment]
> if you need pandas add it to the install_requires list in setup.py to fix the CI error

I don't think that we should depend on pandas. Is it easy to do without?

**Comment by jeromedockes** [comment]
Gael Varoquaux <notifications@github.com> writes:

> I don't think that we should depend on pandas. Is it easy to do without?

as far as I can tell it is only used to check input type, to accept
dataframes and series. I guess we can remove it and only accept arrays,
or instead check the presence of an attribute `to_numpy`

**Comment by GaelVaroquaux** [comment]
> or instead check the presence of an attribute `to_numpy`

The later.

**Pull request closed without merging**